tests/test_functions.py

```diff
@@ -619,3 +619,20 @@ def test_round_to_nearest_half(dataframe):
     assert df.iloc[6, 1] == 1.0
     assert df.iloc[7, 1] == 2.5
     assert df.iloc[8, 1] == 3.0
+
+
+def test_make_currency_api_request(dataframe):
+    """This test assumes that the GBP will always be more than the USD."""
```

Comment:
From a testing engineering perspective, this is a very big assumption to make, even though it is reasonable in our daily experience.

I wonder what you might think about the following test instead?

This test ensures that the API provider is live. This way, in automated nightly builds, if it goes down, we can get notified. It might be good to have a backup exchange rate API as well. 

```python
r = requests.get('https://api.exchangeratesapi.io')
assert r.status_code == 200
```

This one might be the most crucial for the functionality of `convert_currency`.

Reply:
I had the same concerns when writing the tests as well. You're right that the 200 check should be sufficient.

Maybe instead of using another API (since this appeared to be one of the few that was totally free) we can hard-code the rates from a certain day and fall back to that if the API fails. For example, use something like 'exchange rate on first day of most recent quarter' or 'average exchange rate over the past week'. 

Or we can still use the above hard-coded solution and use other APIs as well. We can add a parameter for people to choose which method/API is required and have an order of succession if they fail.

Reply:
> Maybe instead of using another API (since this appeared to be one of the few that was totally free) we can hard-code the rates from a certain day and fall back to that if the API fails.

Not a bad idea. What about this other idea? Provide a function that caches the currency exchange rates for just the current day, which can then be read if offline? That would be more work, though, and I'd prefer for it to be a separate PR, to give us more time to think about it, and also prevent this one from mushrooming. :smile:.

In response to my own thought:

> It might be good to have a backup exchange rate API as well.

On second thought, this might not be a good idea, since this the one you've put up is a free service. To keep things simple, it may be better to just use this API for now.